xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/search/searchSuggest.js

```diff
@@ -118,7 +118,9 @@ var XWiki = (function (XWiki) {
      * Callback triggered when the suggest element is collapsed, because of a focusout event for example.
      */
     onSuggestCollapsed: function(event) {
-      this.suggest.clearSuggestions();
+      /* We match the collapse of the input field timing to close the suggest panel.
+         As of 16.2.0-RC1, this value is defined in action-menus.less -> #headerglobalsearchinput */
```

Comment:
Thanks for the doc improvment. Just to be sure: there's no way to actually extract the data from the CSS rule with some javascript? e.g. in xwiki.org I'm able to get it using `jQuery('#headerglobalsearchinput').css('transition-duration')` (not sure it works with all browsers tbh). 

My point here is I'm afraid we forget to update this value if we ever change the less file and it can also be modified for having a custom skin, which would make that code not efficient.

Reply:
Thanks for the comment! I didn't know JQuery offered this feature. It seems to work well on recent FF and Chromium.
I updated it in https://github.com/xwiki/xwiki-platform/pull/2835/commits/0531d28a63b1222a2fdc3e27d2ae2953466a42d7 

I never even looked for such a feature because it looked to me like a violation of the MVC principle. Having the object behave in a different way only based on its style didn't sound right to me. In this use case it's convenient and somewhat appropriate :+1: